### ISSUE TITLE
Balance Changes

### ISSUE THREAD

**Issue by MitBoy42** [JeffAdriaanse/The-Legend-of-Bum-bo-Windfall#30]
First of all, amazing mod that finally makes the game actually playable! So far  have cleared the Wise and did 5 streak with The Lost, so 8 runs in total.

While I agree with all the balance changes Ive seen so far, there are still issues I see with decent amount of spells. (It's honestly incredible how bad Ed is with balancing of non-platformer or action games).
1. Juiced is still really strong and probably should be a 6 cost.
2. Pill that gives you 1 damage for a turn is very strong and probably should be a 3 or 4 mana cost.
3. 20/20 is really strong and probably should be 5 or 6 mana cost.
4. Buzz up and Down are absolutely OP to the point of being almost run winning and probably should be 2 mana cost each 
5. Time Walk and Wooden Spoon probably should have their cost reduced even more? Perhaps since Im playing as the Lost I always gravitate towards less costly spells but I don't find them worth taking ever. 
6. I have not used this trick in the mod so far and maybe you removed it already, but 7 tile hear combo should not give you a soul heart as it encourages to just spend half an hour in a room with one enemy once you find mom's lipstick or other such item.

**Comment by JeffAdriaanse** [comment]
I'm glad to hear you're enjoying the mod!

I actually did have Juiced! cost 6 mana and 20/20 cost 7 mana in an earlier version of the mod. I have since tried to focus on buffing weak spells rather than nerfing strong ones (I think this is Edmund's typical approach with Isaac), so I mostly reverted those changes. I understand the sentiment though, they are very powerful spells. I also agree with you about Addy and the Buzz spells, and Time Walker and Wooden Spoon could be a bit cheaper.

I am aware of the Mega Love tile combo cheese. Bum-bo can grind coins and health with a variety of spells. I have a balance patch on my to-do list that addresses the issue by making heart tiles turn grey and stop giving health after enough turns have passed. The patch also adds a turn timer to spells that give resources directly, such as Coin Roll and Prayer Card, which disables them after several turns have passed. Some of them would also have their mana costs reduced to compensate.

**Comment by MitBoy42** [comment]
Sounds cool! Another thing I noticed specifically for The lost, is that battery items (ones that give movement) are really OP on him, as you can stack up charges on them with ghost tiles, and get move movements, then do more matches, get more ghost tiles, and so on. 
Im not sure how to fix this really, but maybe a ghost tile match can only charge one item at a time, so it would rechange The negative first, and only if it is charged, recharge the next item (possibly then 5 or 6 tile combos can give more charges at a time).

**Comment by MitBoy42** [comment]
Also the backstabber doesnt seem  to crit on longits (the ceiling worms), perhaps because they hide upon taking damage.

**Comment by JeffAdriaanse** [comment]
That is a good way to balance the Lost. I could also picture a system where he can store the charges and then choose which spell to use them on.

Good catch with the Longits Backstabber bug. I'll see about fixing that in the next update.

**Comment by YazawaAkio** [comment]
oh hi mitboy

**Comment by MitBoy42** [comment]
A few other notes:
1. Compost, while fun, is definitely the strongest trinket in the game by far. Feels like it only makes sense for the Wise as he has one move. Probably should be 10-15% chance otherwise. 
2. Impostor  (2nd chapter enemy) seems rather strange, being almost the same as greedling, I feel like he is supposed to summon a leaper on death? Same with Tato Dad.
3. Dusc and Tainted Dusk have strange interaction with boogers - despite having 4 moves, 2 boogers cancel all 4 moves. 
4. Black lotus and Trinkets that give mana of specific colour probably should be buffed. I tried a run specifically dedicated to using 2 Lotus and it barely worked even on the lost. 
5. I wonder if some spells are not available in the first chapter. I don't think I ever got a brimstone or krampus cross in Dross. If it's true, im not sure if locking spells behind specific chapters is a good idea as you tend to have a certain set build already by the time you get to late game so you are less likely to take them. 
6. Lucky finger spell that places a wild probably can have be a bit more expensive, like 5 mana.
7. Skewer is one of the weaker puzzle spells and probably can be 3 mana. 

Other than that, I enjoy the game so far! This is not a game that requires a skilled artist, but in case you need help with spriting, I can help with drawing stuff!

**Comment by JeffAdriaanse** [comment]
I agree that Compost Bag is too powerful on characters besides the Wise. Making it chance based is a good way to balance it.

There are some harder rooms that have Imposters or Daddy Tatos in them. I'd prefer not to make those enemies spawn Leapers on death because it might make those rooms too difficult. It feel like the game was balanced around Imposters and Daddy Tatos as they are now.

Speaking of which, there actually is some unused code from the vanilla game that would give a few enemies extra on death effects. The code would cause Squats to spawn a Dip on death, Cursers to place 3 curse tiles on death, and Masks to spawn an Imposter on death.

I am aware of the special effect where two boogers will usually cancel all of an enemy's moves. It has been in the back of my mind for quite a while. I feel like poop barriers are usually more effective than boogers as a form of protection, so I initially assumed the effect was an intentional game mechanic to bring boogers more in line with poop barriers in terms of usefulness. I could never find the code where the effect was explicitly programmed in though. I have since looked at the code more closely and I can confirm it is a bug. The fix is straightforward so I should have that ready in the next update. Although I am tempted to deliberately program the effect into the game as a way to make boogers more useful.

I agree that Dark Lotus is underwhelming. I also agree that the mana trinkets need buffs. I'm thinking of doubling the amount of mana granted by Chicken Bone, False Teeth, Toilet Seat, Turdy, Used Tissue, and Small Box.

It makes sense that you haven't encountered Brimstone in the Sewers of Dross. This is because the spell types offered in each treasure room are [predetermined](thelegendofbumbo.fandom.com/wiki/Treasure_Room). It so happens that neither of the treasure rooms in the first chapter contain an attack spell. I have considered making the spell types random in each treasure room, so that might happen in a future update. Aside from that, there is no code restricting spells to specific chapters.

**Comment by MitBoy42** [comment]
Gotcha! By the way, on my channel Ive been uploading my Bumbo the lost streaks. https://www.youtube.com/@MitBoy_/videos
I also might switch to an all character streak when I die, although ive done 20+ runs and it has not happen yet. 

Another thing I tend to notice I have never used a reroll spell syringe and reroll mana cost syringe. Reroll spell seems worthless as you can just find more spells later and reroll mana one feels too risky. 

Are you planning to introduce more content into the game such as rooms or enemies?

**Comment by JeffAdriaanse** [comment]
Your videos are great! Impressive win streak with the Lost.

Good catch with the Pinky bug where it doesn't make wild tiles, I will fix that in the next update. Also, Rainbow Tick will work on 3 mana cost spells again.

I will probably make the Randomize Needle cheaper so that it is more worth using. I'll see about buffing the Shuffle Needle, although I don't have a specific change in mind at the moment. Silver Skull granting 2 mana is a good idea.

I have an update in mind that will add a new chapter to play through. It will involve new rooms types, room layouts, and enemies. I am making a balance update first, the new content will probably be in the next update after that.

**Comment by MitBoy42** [comment]
Cool! In case you need an artist for free, i can help with art!

**Comment by JeffAdriaanse** [comment]
New update is out! Most of the balance problems and bugs have been addressed. I changed Randomize Needle and Shuffle Needle to show the new spell/mana cost before you commit to the purchase, which I think makes them much better upgrades. I'm interested to hear what you think.
